index.rst

```diff
@@ -1,51 +1,1166 @@
+            /* Operator code here */
+        }
+
+        /* Code not needed for wrapping the current function here */
+    };
+
+So we begin with ``Function`` in ``afw/math/FunctionLibrary.h`` by adding ``'functionLibrary'`` to ``afw/math/SConscript``, ``from ._functionLibrary import *`` to ``mathLib.py``, and ``#include "lsst/afw/math/FunctionLibrary.h"`` in ``functionLibrary.cc`` just like we did for ``minimize.h`` in :ref:`new_cpp`.
+
+Below ``using namespace lsst::afw::math;`` and before ``PYBIND11_PLUGIN`` we create the new template function
+
+.. code-block:: c++
+
+    template <typename ReturnT>
+    void declarePolynomialFunctions(py::module &mod, const std::string & suffix) {
+    };
+
+where ``suffix`` will be a string that represents the return type of the function ("D" for double, "I" for int, etc.).
+Inside the function we declare our class
+
+.. code-block:: c++
+
+        py::class_<PolynomialFunction2<ReturnT>, BasePolynomialFunction2<ReturnT>>
+            clsPolynomialFunction2(mod, ("PolynomialFunction2" + suffix).c_str());
+
+This is slightly different than our class declaration in :ref:`wrap_struct` because ``PolynomialFunction2`` inherits from ``BasePolynomialFunction2``, which can be seen in the above declaration.
+Since ``BasePolynomialFunction2`` is defined in ``Function.h`` we must ``#include "lsst/afw/math/Function.h"`` at the beginning of ``functionLibrary.cc``.
+We will discuss inheritance more in :ref:`wrapping_inheritance`.
+Also notice that we combine ``PolynomialFunction2`` with the suffix, specified when ``declarePolyomialFunctions`` is defined, that specified the type for the function (for example "D" or "I").
+
+We notice that the constructor is overloaded, so we define ``init`` with both sets of parameters
+
+.. code-block:: c++
+
+    clsPolynomialFunction2.def(py::init<unsigned int>());
+    clsPolynomialFunction2.def(py::init<std::vector<double> const &>());
 
-.. sectnum::
 
-.. Add content below. Do not include the document title.
+We must also declare the classes in the module, so inside ``PYBIND11_PLUGIN`` and beneath the module declaration ``py::module mod("_functionLibrary", "Python wrapper for afw _functionLibrary library");`` we add
+
+.. code-block:: c++
+
+    declarePolynomialFunctions<double>(mod, "D");
+
+where we use the ``double`` type since ``PolynomialFunction2D`` is the method called from ``testMinimize.py`` and specify ``suffix`` as ``"D"``.
+
+The last piece to wrap in ``functionLibrary.cc`` is the ``__call__`` method, since ``testMinimize.py`` makes use of it.
+Most operators can be wrapped with the helpers in ``pybind11/operators.h``, but for function call we need to specify the operator
+ourselves by binding a lambda.
+
+.. code-block:: c++
+
+    clsPolynomialFunction2.def("__call__", [](PolynomialFunction2<ReturnT> &t, double &x, double &y)
+        -> ReturnT {
+            return t(x,y);
+    }, py::is_operator());
```

Comment:
Looking back at afw__math it looks like I did fix this later and removed the lambda. I'l fix the tutorial to fix this, since it will be nice to show the proper way to wrap operator(), and add another example that uses a lambda function, as well as a python extension.